pyvium/tools/data_processing_functions.py

```diff
@@ -8,17 +8,21 @@
 
 class DataProcessing:
     @staticmethod
-    def _extract_data_section(lines: List[str], start_index: int) -> List[List[int]]:
-        """Extracts a section of data from a list of lines starting at the given index. """
-        section_data = []
-        num_points = int(lines[start_index + 2])
-        start = start_index + 3
-        end = start + num_points
-        for line_index in range(start, end):
-            row = lines[line_index].split()
-            datapoint = [eval(value) for value in row]
-            section_data.append(datapoint)
-        return section_data
+    def _extract_data_section(lines: List[str], start_index: int) -> List[List[float]]:
+        """Extracts a section of data from a list of lines starting at the given index."""
+        try:
+            section_data = []
+            num_points = int(lines[start_index + 2].strip().replace('\x00', ''))
+            start = start_index + 3
+            end = start + num_points
+            for line_index in range(start, end):
+                row = lines[line_index].split()
+                datapoint = [float(value) for value in row]
+                section_data.append(datapoint)
+            return section_data
+        except ValueError as e:
+            print(f"Warning: Could not parse data section. Error: {e}")
+            return []
```

Comment:
Not sure if this is a good way to handle errors. Why not throw an exception?

Reply:
Throwing an exception will stop the program from executing. If there's an unexpected input location for any of the data, I want to make all the data that's possible to recover recoverable.